File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.java
import java.util.*;
class Solution {
    public int solution(int n, int m, int[] section) {
        int answer = 0;
        // 일단 section 마다 다 칠하면 간단하겠지만 한번 칠하는 영역에 이전에 칠했던 영역을 제외하면 더 좋겠다.

        int start = section[0];
        answer++;
        
        for (int sec : section) {
            if (start + m > sec) {
                continue;
            }
            
            start = sec;
            answer++;
        }
        return answer;
    }
}

File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.py
def solution(n, m, section):
    answer = 0
    edge = 0    
    
    for sec in section:
        
        if sec > edge:
            edge = sec + m - 1
            answer += 1
        
        
    return answer

'''
칠했다는 표시를 기록할 리스트 필요?
[true, false, false, true, true, flase, true, true]

칠할 수 있는 영역 한계는 n-m+1

section 원소 기준으로 m-1 뒤칸까지 안칠해진 부분이 있거나
칠할 수 있는 영역 한계 내에 있는 경우 칠한다.

'''